### ISSUE TITLE
Alt text validation rule in the accessibility checker

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#11967]
### Is your proposal related to a problem?

Alt text in Wagtail needs improvements :) For background, see [RFC 89: ATAG 2.0 audit](https://github.com/wagtail/rfcs/pull/89), [RFC 51: Contextual alt text](https://github.com/wagtail/rfcs/pull/51), and the [Alt text capabilities GSoC project idea](https://github.com/wagtail/gsoc/blob/main/project-ideas.md#alt-text-capabilities).

### Describe the solution you'd like

A very practical improvement we can make is to add a new pass/fail rule in our accessibility checker, checking the contents of images’ alt text, against patterns which are guaranteed to provide no false positives.

### Describe alternatives you've considered

We’ve considered a lot of alternatives as part of the different initiatives linked above. Some have the potential to be even more radical improvements, but they’re not necessarily backwards-compatible or designed to help with existing content. An accessibility checker rule is a great way to improve the accessibility of existing content.

### Additional context

We will need to do this with the [axe.configure API](https://github.com/dequelabs/axe-core/blob/master/doc/API.md#api-name-axeconfigure).

Here is a quick prototype:

```js
function checkImageAltText(node: Element) {
      const image = node as HTMLImageElement;
      const altText = image.getAttribute('alt') || '';
      const hasFileExtension = altText.includes('.png');
      return !hasFileExtension;
    }

    const imageAltNoFileExtensionRule = {
      id: 'image-alt-no-file-extension',
      impact: 'serious' as ImpactValue,
      selector: 'img[alt]',
      tags: ['best-practice'],
      any: ['check-image-alt-text'],
    };

    axe.configure({
      checks: [
        {
          id: 'check-image-alt-text',
          evaluate: checkImageAltText,
        },
      ],
      rules: [imageAltNoFileExtensionRule],
    });
```

### Working on this

Assigned to @albinazs. If anyone else wants to help with this, feedback is very welcome in comments in this issue.

**Comment by thibaudcolas** [comment]
I have prepared two datasets of images alt text which we can use to trial our new rule:

- [35k alt text of images](https://docs.google.com/spreadsheets/d/1MgHVWYuKjdLkqiCOJzli7wQKjmjpeHjLL1naJYHeeDw/edit#gid=2009276316) –  displayed on Wagtail websites’ homepages
- [19k alt text of CMS-managed images](https://docs.google.com/spreadsheets/d/1MgHVWYuKjdLkqiCOJzli7wQKjmjpeHjLL1naJYHeeDw/edit#gid=303545789) – also from Wagtail websites’ homepages

See [thibaudcolas/alt-text-benchmark](https://github.com/thibaudcolas/alt-text-benchmark) for a reference on how this data was produced.

**Comment by thibaudcolas** [comment]
Merged :) Created #12091 as a follow-up.
